### PR TITLE
Automated Workflow Executions UI

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in the UI for configuring automated workflow executions for a Project. Currently this includes listing, adding, and deleting of automated workflow executions.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Run server
2. Navigate to a Project where you are `Owner` or `Maintainer`
3. Click on `Settings` -> `Automated Workflow Executions`
4. Click on `Add new automated workflow execution`
5. Click on an automatable pipeline (if list is empty ensure that your `config/pipelines/pipelines.json` has at least one pipeline version with `"automatable": true`, refer to default file in repo)
6. Click on `Update samples with analysis results` and `Receive an email notification when your analysis has completed?`
7. Click on `Submit`
8. Add a new Sample to the Project
9. Upload pair end data to the new Sample
10. Wait for workflow execution to be ran and confirm that files and metadata have been added to the Sample

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/components/nextflow_component.html.erb [hunk 1]
```diff
@@ -135,7 +137,11 @@
       </div>
     <% end %>
 
-    <button type="submit" class="button button--state-primary button--size-default" data-nextflow--samplesheet-target="submit">
+    <button
+      type="submit"
+      class="button button--state-primary button--size-default"
```
Comment (joshsadam) [inline app/components/nextflow_component.html.erb:142]: **nitpick:** We are trying to get ride of these classes as the `@apply` rule is frowned upon (apparently). Can you just use the class lists directly from flowbite.

File: app/views/projects/automated_workflow_executions/_pipeline_selection_modal.html.erb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          <%= link_to new_namespace_project_automated_workflow_execution_path(@project.parent, @project, workflow_name: workflow.name, workflow_version: workflow.version), class: "group w-full", data: { turbo_stream: true } do %>
+            <span
+              class="
+                ws-default
+                flex
+                items-center
+                p-4
+                space-x-4
+                hover:bg-primary-600
+                btn
```
Comment (joshsadam) [inline app/views/projects/automated_workflow_executions/_pipeline_selection_modal.html.erb:16]: **nitpick:** Can you replace `btn` with the actual flowbite classes.  Where did you copy that from?

**Review by joshsadam — CHANGES_REQUESTED** [review]
Just a couple general comments, this is looking really good

File: app/components/nextflow_component.html.erb (hunk 1 above)
Comment (ericenns) [reply app/components/nextflow_component.html.erb:142]: I'd rather leave this for now as this was just my IDE reformatting this line, vs me actually refactoring this portion.

File: app/views/projects/automated_workflow_executions/_pipeline_selection_modal.html.erb (hunk 2 above)
Comment (ericenns) [reply app/views/projects/automated_workflow_executions/_pipeline_selection_modal.html.erb]: Removed in f9dea938a9deda9d83d0218f1b7a935f247bc25a

File: app/views/projects/automated_workflow_executions/_pipeline_selection_modal.html.erb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                  classes:
+                    "ws-default dark:text-slate-500 group-hover:text-slate-100 group-hover:-rotate-45 transition-rotate-45 group-hover:duration-500 group-hover:ease-in-out"
+                ) %>
+              </span>
+              <span class="flex-1 min-w-0 text-left group-hover:stroke-slate-100">
+                <span class="flex items-center space-x-2">
+                  <span
+                    class="
+                      font-medium
+                      text-gray-900
```
[the remark this answers is not in the document]
Comment (ericenns) [reply app/views/projects/automated_workflow_executions/_pipeline_selection_modal.html.erb]: Removed in 431216819972cae290b81a1f61ee01485c0efbb2

File: config/locales/en.yml [hunk 4]
```diff
@@ -824,6 +824,35 @@ en:
...
+        title: Automated Workflow Executions
+        subtitle: Setup automated workflow executions to launch when pair end data is uploaded to the project
+        add_new_automated_workflow_execution: New automated workflow execution
+      actions:
+        delete_button: Delete
+        delete_confirm: Are you sure that you want to delete this automated workflow execution?
+      table:
+        empty:
+          title: No automated workflow executions
+          description: There are no automated workflow executions configured for this project
```
[the remark this answers is not in the document]
Comment (ericenns) [reply config/locales/en.yml]: Fixed in 580938e176894fe2cd20835636f4dc0b32e6d6ed

File: app/components/nextflow_component.html.erb (hunk 1 above)
Comment (joshsadam) [reply app/components/nextflow_component.html.erb:142]: Sounds good

File: app/views/projects/automated_workflow_executions/_pipeline_selection_modal.html.erb (hunk 2 above)
Comment (joshsadam) [reply app/views/projects/automated_workflow_executions/_pipeline_selection_modal.html.erb]: Thanks

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Review by joshsadam — APPROVED** [review]
This looks good to me

**Pull request merged**